### PR TITLE
Decorate a class

### PR DESCRIPTION
**Pull request by AbrahaKahsay**
# Project description
In this project, I have used the Decorator design pattern to correct the names of people.

## Project requirements

- Check the usage of [Decorator pattern in Ruby](https://refactoring.guru/design-patterns/decorator/ruby/example#example-0)
- Think about how you can use two decorators in order to capitalize and trim people's names.
- In the following steps you will get instructions for implementing the Decorator design pattern for this case.
- Create a class Nameable.
    - Implement a method called correct_name that will raise a NotImplementedError.
 
**Turn your Person class to Nameable**

- Make sure that your Person class inherits from Nameable
- Make sure that this class has a method correct_name implemented. It should simply return the name attribute.

**Prepare base Decorator**

- Make sure that it inherits from Nameable.
- In the constructor assign a nameable object from params to an instance variable.
- Implement the correct_name method that returns the result of the correct_name method of the @nameable.

**Prepare CapitalizeDecorator and TrimmerDecorator**

- For the CapitalizeDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method correct_name that capitalizes the output of @nameable.correct_name.
- For the TrimmerDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: capitalize_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require './base_decorator'
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    super().upcase
+  end
+end
```
Comment (BenMukebo) [inline capitalize_decorator.rb]: - [ ] Great job so far 👏🏻 However you should Implement a method correct_name that `capitalizes` the output of @nameable.correct_name and not `upcase`. It should look similar to:

```ruby
class CapitalizeDecorator < Decorator
    # Capitalize the first letter of the given name.
    def correct_name
      @nameable.correct_name.capitalize
    end
end
```

- Your output

**Review by BenMukebo — APPROVED** [review]
### Hi @AbrahaKahsay,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- Gitflow was correctly used.  ✔️   
- Descriptive README file.  ✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AbrahaKahsay** [comment]
@BenMukebo Thanks! Great catch and great lesson! I will fix that.

**Review by brytebee — APPROVED** [review]
Hi, @AbrahaKahsay👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great schema design.✔️
- App functions as it should.✔️

##

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



Good luck! 🎉 

##

###  **Optional Suggestions:**
- N/A

------

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**